wapitiCore/attack/mod_csrf.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python3
+from collections import Counter
+import math
+from requests.exceptions import ReadTimeout
+
+from wapitiCore.attack.attack import Attack
+from wapitiCore.language.vulnerability import Vulnerability, Anomaly, _
+from wapitiCore.net.web import Request
+from wapitiCore.net.crawler import Page
+
+class mod_csrf(Attack):
+    """
+    This class implements a CSRF/XSRF attack
+    """
+
+    name = "csrf"
+
+    do_get = True
+    do_post = True
+
+    csrf_string = None
+
+    MIN_ENTROPY = 3
+    MIN_LENGTH = 8
+
+    TOKEN_FORM_STRINGS = [
```

Comment:
I don't know what is the "academic" way to check for CSRF tokens but known names seems good.

Using entropy() on form fields (names) may be a way to discover dynamic CSRF tokens but there will be risk of more false positives of course.

Reply:
Known names seems ok on websites developed with known frameworks but it is limited since it is recommended to use custom names for csrf token fields. 

Several tokens won't be detected as it is, so many false positives will be raised. I don't know how to ensure that a field is a csrf token when it is a custom name. I don't understand how entropy() could help on that.